prometheus_client/gc_collector.py

```diff
@@ -67,15 +62,66 @@ def _cb(phase, info):
 
                 if phase == 'stop':
                     delta = time.time() - times[gen]
-                    latency.labels(gen).observe(delta)
-                    if 'collected' in info:
-                        collected.labels(gen).observe(info['collected'])
-                    if 'uncollectable' in info:
-                        uncollectable.labels(gen).observe(info['uncollectable'])
+                    self._add_to_bucket(self._LATENCY, gen, delta)
+                    if 'collected' in info and info['collected'] != 0:
+                        self._add_to_bucket(self._COLLECTED, gen, info['collected'])
+                    if 'uncollectable' in info and info['uncollectable'] != 0:
+                        self._add_to_bucket(self._UNCOLLECTABLE, gen, info['uncollectable'])
             finally:
                 self.gc_cb_active = False
 
         gc.callbacks.append(_cb)
+        registry.register(self)
+
+    def _add_to_bucket(self, bucket_name, gen, value):
+        bucket = self._buckets[gen][bucket_name]
+        self._sums[gen][bucket_name] += value
+
+        for bound in self._get_bounds(gen, bucket_name):
+            if value <= bound:
+                bucket[INF] += 1
+                bucket[bound] += 1
+                break
+
+    @staticmethod
+    def _get_bounds(gen, bucket_name):
+        if bucket_name == GC_COLLECTOR._LATENCY:
+            return .005, .01, .025, .05, .075, .1, .25, .5, .75, 1.0, 2.5, 5.0, 7.5, 10.0
+        _max = gc.get_threshold()[gen] * 2
```

Comment:
Could this vary over time?

It'd also be best to have the same buckets for each generation, otherwise you can't aggregate them.

Reply:
In general it is possible to change threshold.  
As a solution we can take threshold of 0 generation (the biggest one) as max border